compliance_checker/cf/cf_1_6.py

```diff
@@ -1954,24 +1954,15 @@ def check_standard_calendar_no_cross(time_var):
             Check that the time variable does not cross the date
             1582-10-15 when standard or gregorian calendars are used
             """
-            # IMPLEMENTATION CONFORMANCE 4.4.1 RECOMMENDED 2/2
-            # Only get non-nan/FillValue times, as these are the only things
-            # that make sense for conversion.  Furthermore, non-null checks
-            # should be made for time coordinate variables anyways, so errors
-            # should be caught where implemented there
-            crossover_date = cftime.DatetimeGregorian(1582, 10, 15)
-            # has_year_zero set to true in order to just check crossover,
-            # actual year less than or equal to zero check handled elsewhere
-            # when standard/Gregorian, or Julian calendars used.
-
             # WARNING: might fail here if months_since are used and suppress
             #          usual warning
             try:
-                times = cftime.num2date(
+                cftime.num2date(
                     time_var[:].compressed(),
                     time_var.units,
                     has_year_zero=True,
                 )
+                time_values = time_var[:]
```

Comment:
If we catch this elsewhere we can probably remove this section.

Reply:
We do catch this elsewhere too. We can fail early here avoid creating the `cftime` object speeding this up to < 1 s.